DTS-STN/canadian-dental-care-plan
# Adding benefit application entity and mapper and defining separate types for each dto field

### Description
Incremental PR to make benefit application service injectable, which includes adding:
- benefit application entity based off of [benefit-application-service-schemas](https://github.com/DTS-STN/canadian-dental-care-plan/blob/main/frontend/app/schemas/benefit-application-service-schemas.server.ts)
- benefit application DTO mapper based off of [benefit-application-service-mappers](https://github.com/DTS-STN/canadian-dental-care-plan/blob/main/frontend/app/mappers/benefit-application-service-mappers.server.ts)
- separate types for benefit application DTO, allowing the DTOs to be:
    - referenced as argument types in the mapper 
    - extended from the session state types in the future

Minimal code refactoring was done in the mapper to reduce risk of unintended behavior changes or errors in the mapping logic.

### Checklist

### Test Instructions
Inject and use the DTO mapper into any `/review-information` screen for the `/apply` flow where the `view-payload` feature is enabled and verify that the expected payload is outputted. 

For example, in `/apply/$id/adult/review-information`, modify the `payload` initialization in the `loader(..)`:
```typescript
const benefitApplicationDto = appContainer.get(TYPES.domain.mappers.BenefitApplicationStateMapper).mapApplyAdultStateToBenefitApplicationDto(state);
const payload = viewPayloadEnabled && appContainer.get(TYPES.domain.mappers.BenefitApplicationDtoMapper).mapBenefitApplicationDtoToBenefitApplicationRequestEntity(benefitApplicationDto);
```
